### PR TITLE
cli: Use Apollo's new multi-device print_info

### LINKED ISSUE #125: Flash UID Zeros After --force-offline
After running `cynthion info --force-offline` the Flash UID shows up only as 0's.  Performing a reset and running `cynthion info` the Flash ID is displayed correctly again.

Hardware: Cynthion r1.4
Firmware version: v1.0.6
USB API version: 1.1
PyPi cynthion version: 0.1.0

Workaround: To reliably find your Cynthion's flash UID, use `apollo flash-info`.

### LINKED ISSUE #142: `cynthion update` unreliable when enumeration is slow
There are some fixed delays at multiple points in the process that are not long enough for all host computers. It would be better to wait until the device appears on the bus and then give up only after a (longer) timeout.

### PR DESCRIPTION
**Pull request by mossmann**
Also update dependencies to Apollo 1.1.0 and pyfwup 0.5.0.

Fix #125
Fix #142

### CODE REVIEW

File: cynthion/python/src/commands/cynthion_info.py [hunk 1]
```diff
@@ -2,32 +2,22 @@
...
-
-    # tell user how to get all display info if we weren't in Apollo mode
-    if device is None:
-        logging.info("\nApollo stub interface found. Press the PROGRAM button to switch the device")
-        logging.info("to Apollo mode or use 'cynthion info --force-offline' to show all device info.")
+def cynthion_info(args):
+    logging.info(f"Cynthion version: {__version__}")
+    if ApolloDebugger.print_info(force_offline=args.force_offline, out=logging.info,
+                ids=[(shared.usb.bVendorId.apollo, shared.usb.bProductId.apollo)],
+                stub_ids=[(shared.usb.bVendorId.cynthion, shared.usb.bProductId.cynthion)]):
```
Comment (antoinevg) [inline cynthion/python/src/commands/cynthion_info.py:21]: 👆 _Nice._

File: cynthion/python/src/__init__.py [hunk 2]
```diff
@@ -20,6 +25,9 @@
 from . import gateware
 from . import shared
 
+import importlib.metadata
+__version__ = importlib.metadata.version(__package__)
```
Comment (antoinevg) [inline cynthion/python/src/__init__.py:29]: `importlib` always scares me in the particular range of python versions we operate in, has this been tested on a clean Python 3.8 install?

I vaguely remember having to do a `from importlib.metadata import version` or add a: `"importlib-metadata; python_version<'3.9'",` to the project dependencies but that could have been on Python 3.7…

File: cynthion/python/src/commands/cli.py [hunk 3]
```diff
@@ -84,29 +83,12 @@ def main():
...
-            sys.exit(1)
-        else:
-            # device is connected but we are not in apollo mode
-            device = None
 
     # Execute the relevant command.
     if args.func is cynthion_info:
-        # FIXME ugly hack to handle bitstream info when not in Apollo mode
-        args.func(device, args, bitstream_info)
+        args.func(args)
```
Comment (antoinevg) [inline cynthion/python/src/commands/cli.py:91]: 🎉

File: cynthion/python/src/commands/cli.py [hunk 4]
```diff
@@ -84,29 +83,12 @@ def main():
...
-    else:
-        bitstream_info = get_bitstream_information()
-
     # Force the FPGA offline by default in most commands to force Apollo mode if needed.
     force_offline = args.force_offline if "force_offline" in args else True
-    try:
-        device = ApolloDebugger(force_offline=force_offline)
-    except Exception as e:
-        # FIXME ugly hack to handle bitstream info when not in Apollo mode
-        if bitstream_info is None:
```
Comment (antoinevg) [inline cynthion/python/src/commands/cli.py:99]: 🎉

File: cynthion/python/src/commands/cli.py [hunk 5]
```diff
@@ -84,29 +83,12 @@ def main():
         parser.print_help()
         return
 
-    # Retrieve information about the active bitstream (if any).
-    if hasattr(args, "force_offline") and args.force_offline:
-        bitstream_info = None
-    else:
-        bitstream_info = get_bitstream_information()
-
```
Comment (antoinevg) [inline cynthion/python/src/commands/cli.py:92]: 🎉

File: cynthion/python/src/commands/util.py [hunk 6]
```diff
@@ -26,34 +25,6 @@
...
-    major = device.bcdDevice >> 8
-
-    return {
-        "product": device.product,
-        "manufacturer": device.manufacturer,
-        "hardware": f"r{major}.{minor}",
-        "flash_uid": device.serial_number,
-    }
-
-
```
Comment (antoinevg) [inline cynthion/python/src/commands/util.py:56]: 🎉

File: cynthion/python/src/__init__.py (hunk 2 above)
Comment (mossmann) [reply cynthion/python/src/__init__.py:29]: Yes, `importlib.metadata` has been included in stdlib since 3.8. I've tested with 3.8.

Comment (mossmann) [reply cynthion/python/src/__init__.py:29]: You're probably remembering what you did with `importlib.resources`, not `importlib.metadata`.

**Pull request merged**